modules/engage-paella-player/pom.xml

```diff
@@ -0,0 +1,138 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>opencast-engage-paella-player</artifactId>
+  <packaging>bundle</packaging>
+  <name>Opencast :: engage-paella-player</name>
+  <parent>
+    <groupId>org.opencastproject</groupId>
+    <artifactId>base</artifactId>
+    <version>5-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+  <properties>
+    <opencast.basedir>${project.basedir}/../..</opencast.basedir>
+    <checkstyle.skip>false</checkstyle.skip>
+  </properties>
+  
+  <dependencies>
+    <dependency>
+      <groupId>org.opencastproject</groupId>
+      <artifactId>opencast-search-service-api</artifactId>
```

Comment:
I'm curious why opencast-search-service-api is listed as a dependency. Is it to just make sure that the search-service-api jar has been built and exists along with the paella player in the same assembly?

Reply:
Paella uses the search service to get the episodes. It's just to ensure the search service is is loaded on the same server. Maybe is better to change it to `opencast-search-service-impl`? or I can remove it if it's not needed

Reply:
- Since that dependency technically specifies a Java dependency, I would prefer to remove it.
- Dependency versions should be specified in the main `pom.xml` if there is no good reason to do otherwise. For this bundle, specifying a new `maven-bundle-plugin` could make sense since it's not easily updated for all of Opencast. Every other dependency version should be removed.
- We need to add the build number to the bundle.

For these I suggest applying this patch:

- https://gist.github.com/lkiesow/386ce97c2809c41ef9d13116d8971f13#file-paella-player-pom-patch

Additionally, there is a `frontend-no-prebuilt` profile in Opencast which prevents Opencast from downloading node and npm but use the system node instead. Take a look at the admin-ui bundle `pom.xml` for details. We should probably add that profile to this module as well.

Reply:
I think it removing that reference will make it less confusing. If it's not mentioned in the docs, that would be a better place to explain the dependency. The Opencast assemblies makes sure this module is deployed with search modules.